uwh-refbox/src/main.rs

```diff
@@ -295,67 +332,182 @@ fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
             }
         };
 
-        //temporary values for assigning a penalty
-        let black_penalty_count = 0;
-        let white_penalty_count = 0;
+        // Create Vectors for the Black and White Penalty lists
+        let mut black_penalties = vec![];
+        let mut white_penalties = vec![];
+
+        // Sorting Penalties by Time and then by Color
+        if !state.penalties.is_empty() {
+            state.penalties.sort_by(|a, b| a.time.cmp(&b.time));
+
+            for penalty in &state.penalties {
+                if penalty.color == Color::Black {
+                    black_penalties.push(penalty);
+                } else {
+                    white_penalties.push(penalty);
+                }
+            }
+        }
```

Comment:
The outer `if` here is unnecessary. both the commands inside already won't do anything if the vector is empty.